src/service/resources-impl.js

```diff
@@ -1327,6 +1327,13 @@ export class Resources {
     // Breadth-first search.
     if (element.classList.contains('-amp-element')) {
       callback(this.getResourceForElement(element));
+      // Also schedule amp-element that is a placeholder for the element.
+      if (!element.classList.contains('-amp-layout')) {
```

Comment:
Why `if (!element.classList.contains('-amp-layout'))`? Generally, elements themselves decide when placeholder is visible or not. It's typical that placeholder is hidden after layout, but not guaranteed. Otherwise we'd simply support it via global CSS. I think there's no risk of re-discovering placeholder here in all cases. If it's currently hidden - it won't be processed.